src/library_glfw.js

```diff
@@ -1206,6 +1206,9 @@ var LibraryGLFW = {
     window.addEventListener("keypress", GLFW.onKeyPress, true);
     window.addEventListener("keyup", GLFW.onKeyup, true);
     window.addEventListener("blur", GLFW.onBlur, true);
+    window.addEventListener("resize", () => {
+      GLFW.onCanvasResize(window.innerWidth, window.innerHeight);
```

Comment:
Wouldn't window.innerWidth, window.innerHeight be the size of the window rather than the size of the canvas?

Presumably calling `Browser.updateResizeListeners()` would have the intended effect since that would then call the `GLFW.onCanvasResize` which was added to `Browser.resizeListeners`

Reply:
Yeah, I think this is not connect: `onCanvasResize` should be advertising the size of the canvas, not the whole size of the browser window.

There already exists the following code on line 1230/1233 below:

```js
    Browser.resizeListeners.push((width, height) => {
       GLFW.onCanvasResize(width, height);
    });
```
which should do the job? If it is not working then that sounds like some kind of bug in that code.

Note that there are two different sizes at play, because in WebGL the WebGL render target (front buffer) size is decoupled from the size of the visual rectangle that shows up on browser page (the CSS size).

The GLFW resize callback contract is interested in being notified about the WebGL render target size. That is, when a resize is notified, GLFW programs expect that the new render front buffer has the same size that is advertised in the resize callback. So passing CSS size in that callback would not be appropriate.

Reply:
@ypujante is working on  PR at #20584 that affects this. See his examples in the PR description.

Reply:
@pierricgimmig @juj from my experience dealing with this code for a while, this is what I have found:

Working in the context of a browser/canvas is **very** different from a desktop window. For glfw/emscripten, the canvas is the window and it is forcing the size of the canvas via css style (and as @juj mentioned I am working on this and for HiDPI there is no other choice really)

But the best way I have personally found to handle canvas resizing cleanly is to do this:

1) in your html, you wrap your canvas in a `div`:

```html
<div id="canvas-container">
<canvas class=emscripten id=canvas oncontextmenu=event.preventDefault() tabindex=-1></canvas>
</div>
```

then you can apply whatever css you want on the `div`, for example for taking the full window I do this (but you can have margins and whatnot...):

```css
    #canvas-container {
      position: absolute;
      top: 0;
      left: 0;
      padding: 0;
      border: 0 none;
      margin: 0;
      width: 100%;
      height: 100%;
    }
```

2) in your code you register a window size listener on the container (you cannot listen for a resize on a canvas...)

```cpp
  emscripten_set_resize_callback(EMSCRIPTEN_EVENT_TARGET_WINDOW, glfwGetCurrentContext(), false, onCanvasSizeChanged);
```
3) in the callback you handle it this way:

```cpp
static EM_BOOL onCanvasSizeChanged(int /* event_type */, const EmscriptenUiEvent* /* ui_event */, void* user_data)
{
  double canvas_width, canvas_height;
  emscripten_get_element_css_size("#canvas-container", &canvas_width, &canvas_height);
  glfwSetWindowSize(glfwGetCurrentContext(), (int)canvas_width, (int)canvas_height);
  return true;
}
```

Reply:
Got it, it all makes sense now :-) Thanks to all for the replies!